ykrt/src/compile/jitc_yk/aot_ir.rs

```diff
@@ -277,41 +286,20 @@ impl AotIRDisplay for Predicate {
     }
 }
 
-const OPKIND_CONST: u8 = 0;
-const OPKIND_LOCAL_VARIABLE: u8 = 1;
-const OPKIND_TYPE: u8 = 2;
-const OPKIND_FUNC: u8 = 3;
-const OPKIND_BLOCK: u8 = 4;
-const OPKIND_ARG: u8 = 5;
-const OPKIND_GLOBAL: u8 = 6;
-const OPKIND_PREDICATE: u8 = 7;
-const OPKIND_BINOP: u8 = 8;
-const OPKIND_UNIMPLEMENTED: u8 = 255;
-
 #[deku_derive(DekuRead)]
 #[derive(Debug)]
 #[deku(type = "u8")]
 pub(crate) enum Operand {
```

Comment:
When we discussed this design, we were under the impression that `Operand` would have only two variants:
 - constant
 - local

But alas, we can't really do that. Take for example a call instruction. It has a vector of `Operand` and expects to be able to reference all manner of things like globals, functions (for function pointers), arguments to the parent function (which in our IR are distinct from local variables).

The JIT IR works around this by having special opcodes that "convert" anything not a local or a constant to a local. We could have done the same here too, but since we don't have the same space constraints in the AOT IR, I suggest we just keep the slightly larger Operand enum.

Notice that we were however able to delete a lot of other stuff:
 - type
 - block
 - predicate
 - binop
 - unimplemented

Since those things are not values in llvm, they are not operands in our IR.